src/build.rs

```diff
@@ -4,4 +4,9 @@ fn main() {
     if var("CARGO_FEATURE_DEFLATE_MINIZ").is_ok() {
         println!("cargo:warning=Feature `deflate-miniz` is deprecated; replace it with `deflate`");
     }
+    #[cfg(not(any(feature = "sync", feature = "tokio")))]
+    compile_error!("Missing Required feature");
```

Comment:
Update `ci.yml` to enable `sync` wherever it has `--no-default-features`, or else convert it to a `no-sync` feature (which would be unidiomatic but have the advantage of being backward-compatible for more users).

Reply:
Once the tokio feature is properly working in tests I will update the ci. Currently that is broken.

Reply:
Thanks for letting me know; but if it's going to take longer than about another 3 days, then I'd also greatly appreciate a CI-able update once or twice per week so we could get an idea of how the PR was progressing toward readiness.

Reply:
Then I will try doing it this week after I some of my college exams.

Reply:
@SapryWenInera Have you finished your exams? If not, when do you expect to be able to address the comments on this PR? An ETA would be helpful not only for me, but also for the authors of the other major PRs, given the likelihood of merge conflicts.